### ISSUE TITLE
coinstac-simulator fails to run computation

### ISSUE THREAD

**Issue by swashcap** [trendscenter/coinstac#26]
**Problem:** _coinstac-simulator_ hangs when running an computation and declaration _outside_ of the monorepo.

Example repo: https://github.com/swashcap/coinstac-simulator-load-tester ( b4a2a41d1fd2a54cac0ba362cdc516c7d23621bf at the moment).

Explanation:
- _index.js_ is the declaration file. It’s dynamically creates users, and points to…
- _computation.js_ is the decentralized computation. Its `local` and `remote` contain one JavaScript function each. Nothing crazy.
- _coinstac-simulator_ is required and fired off as an NPM script. Run `npm test` to fire things up.
- Observe hanging after the following output:
  
  ``` shell
  info: db server up
  info: db seeded
  info: compute server up
  info: clients up
  ```

**Comment by swashcap** [comment]
Through some carefully placed `console.log`s, I’ve determined that the remote document is never seen by the clients. This loop fires _forever_: https://github.com/MRN-Code/coinstac/blob/master/packages/coinstac-simulator/src/boot-client.js#L90-L106

**Comment by swashcap** [comment]
After placing a logger in _coinstac-simulator/src/boot-client.js_, it’s apparent that no clients are initiating a `RemoteComputationResult` save because the usernames are totally messed up:

```
info: USER sparklingshark [34957]: [ 'obtuseowl', 'gracefulgoat', 'copiouscat' ] 'sparklingshark'
info: USER gorgeousgiraffe [34956]: [ 'rancidrat', 'frightenedfish', 'righteousrat' ] 'gorgeousgiraffe'
info: USER lonelylobster [34958]: [ 'enchantingeagle', 'sluggishspider', 'tensetiger' ] 'lonelylobster'
```

The arrays of usernames (`[ 'obtuseowl', 'gracefulgoat', 'copiouscat' ]`) are different from the client’s username (`'sparklingshark'`). This is because usernames are dynamic in my declaration. Because different Node.js processes are `require`-ing it, the module generates new usernames each time. Thus, username lookups are useless.

**Comment by swashcap** [comment]
Addressed in #23. Dynamic computations now work!
